servicetalk-transport-api/src/main/java/io/servicetalk/transport/api/ConnectionObserver.java

```diff
@@ -28,6 +28,21 @@
  */
 public interface ConnectionObserver {
 
+    /**
+     * Callback when a connection starts initialization and {@link ConnectionInfo} becomes available.
+     * <p>
+     * This callback is invoked before any other callback on this observer.
+     *
+     * @param info {@link ConnectionInfo} for the connection that starts initialization. Note that the
+     * {@link ConnectionInfo#sslSession()} will always return {@code null} since it is called before the
+     * {@link ConnectionObserver#onSecurityHandshake() security handshake} is performed (and as a result no SSL session
+     * has been established). Also, {@link ConnectionInfo#protocol()} will return L4 (transport) protocol.
+     * Finalized {@link ConnectionInfo} will be available via {@link #connectionEstablished(ConnectionInfo)} or
+     * {@link #multiplexedConnectionEstablished(ConnectionInfo)} callbacks.
+     */
+    default void onConnectionInitialization(ConnectionInfo info) {  // FIXME: 0.43 - consider removing default impl
```

Comment:
Open for better name ideas

Reply:
The trick is that at this point `channel.localAddress()` and `channel.remoteAddress()` return `null`. This is not aligned with `ConnectionInfo` API (non-nullable methods).  Another option is to provide `onTransportHandshakeComplete(ConnectionInfo)` instead of `onTransportHandshakeComplete()`. When Channel becomes active, addresses are available.

Reply:
As we discussed offline, changing to `onTransportHandshakeComplete(ConnectionInfo)`